python_dwd/request_dwd.py

```diff
@@ -0,0 +1,226 @@
+
+        assert parameter_name in ["station_id", "parameter", "period_type",
+                                  "time_resolution", "start_date", "end_date"], \
+            "Error: Unknown parameter can not be parsed."
+
+        self.parameter_name = parameter_name
+        self.parameter_value = parameter_value
+
+        # self.extract_parameter_from_value()
+
+    def extract_parameter_from_value(self):
+        parameter_value = self.parameter_value
+
+        if self.parameter_name == "station_id":
+            return self.extract_station_id(parameter_value)
+
+        if self.parameter_name in ["parameter", "period_type", "time_resolution"]:
+            parameter_value = self.clean_string(parameter_value)
+
+            for par in STRING_TO_ENUMERATION_MAPPING[self.parameter_name]:
+
+                wordlist = STRING_TO_WORDLIST_MAPPING[self.parameter_name][par]
+
+                cond1 = len(wordlist) == len(parameter_value)
+
+                cond2 = self.find_any_one_word_from_wordlist(parameter_value, wordlist)
+
+                if cond1 and cond2:
+                    return par
+
+        if self.parameter_name in ["start_date", "end_date"]:
+            return Timestamp(parameter_value)
+
+    @staticmethod
+    def extract_station_id(parameter_value):
+        if all([isinstance(par_val, int) for par_val in parameter_value]):
+            return parameter_value
+
+        if all([isinstance(par_val, float) for par_val in parameter_value]):
+            return [int(par_val) for par_val in parameter_value]
+
+        # Input is expected to be list
+        station_id = []
+
+        for par_val in parameter_value:
+            extracted_station_ids = extract_numbers_from_string(string=par_val, number_type="int", decimal="")
+
+            if len(extracted_station_ids) == 0 or len(extracted_station_ids) > 1:
+                raise ValueError(f"Error: the string {par_val} holds zero or more then one number. This is not "
+                                 f"supported as it's unclear which number to take as station id.")
+
+            station_id.extend(extracted_station_ids)
+
+        return station_id
+
+    @staticmethod
+    def clean_string(string):
+        return string.strip().lower().split("_")
+
+    @staticmethod
+    def find_any_one_word_from_wordlist(string_list, word_list):
+        return all([any([any([(word in string) if not word.isdigit() else word == string
+                        for word in wl]) for string in string_list])
+                   for wl in word_list])
+
+
+class DWDRequest:
+    def __init__(self,
+                 station_id: List[Union[int, str]],
+                 parameter: Parameter,
+                 period_type: PeriodType,
+                 time_resolution: Union[None, str, list, TimeResolution] = None,
+                 start_date: Union[None, str, Timestamp] = None,
+                 end_date: Union[None, str, Timestamp] = None):
+        if not isinstance(station_id, list):
```

Comment:
Please do not ask for permission, ask for sorry afterwards. So the if clause is not necessary

Reply:
I don't see your point. Do you suggest to just call list() on the object? At least if it's a string that could be a problem as:

list("string") -> ["s", "t", "r", "i", "n", "g"]